src/integrator_type.jl

```diff
@@ -55,7 +58,7 @@ mutable struct DDEIntegrator{algType<:OrdinaryDiffEq.OrdinaryDiffEqAlgorithm,
     fsallast::FSALType
 
     # incomplete initialization without fsalfirst and fsallast
-    function DDEIntegrator{algType,uType,tType,P,eigenType,absType,relType,
+    function DDEIntegrator{alg, IIP, uType, tType, P,eigenType,absType,relType,
```

Comment:
The `where` clause has to be updated as well and I guess we have to add the type parameters to `new` as well since `IIP` can not be inferred otherwise - but maybe I'm wrong and the correct signature pf the inner constructor is sufficient?

By the way, I'm wondering whether we should add a constructor that deduces IIP from the given DDEFunction?